src/index.ts

```diff
@@ -47,82 +38,141 @@ async function runTest(files: string[], options: Options) {
+    }
 
-        reporterStream.end();
+    let { testFiles, focusedCasesDetected } = await getTestFiles(execute);
 
-        if (coverageEnabled) {
-            reportCoverage(coverageReporters);
-        }
+    // Fail if we find focused test cases and there weren't supposed to be any.
+    if (forbidFocused && focusedCasesDetected) {
+        let formattedList = testFiles
+            .map((filename) => `\t${filename}`)
+            .join("\n");
+        console.error(
+            c.red(
+                `Error: used command line arg ${c.cyan(
+                    "--forbid-focused"
+                )} but found focused tests in these files:\n${formattedList}`
+            )
+        );
 
-        if (forbidFocused && returnVals.length > 0) {
-            checkForFocusedCases(returnVals);
-        }
-    } catch (e) {
-        console.error("Interpreter found an error: ", e);
-        process.exitCode = 1;
+        return {};
+    }
+
+    testRunner.run(execute, testFiles, focusedCasesDetected);
+    testRunner.reporterStream.end();
+
+    if (coverageEnabled) {
+        reportCoverage(coverageReporters);
     }
 
-    return reporterStream.runner.testResults;
+    return testRunner.reporterStream.runner?.testResults || {};
 }
 
 /**
- * Checks an array of brs elements to see if they have a non-empty focused cases field.
- * @param elements The elements to check for focused cases
+ * Returns the appropriate set of *.test.brs files, depending on whether it detects any focused tests.
+ * Runs through the entire test suite (in non-exec mode) to determine this.
+ * Also returns a boolean indicating whether focused tests were found.
+ * @param execute The scoped execution function to run with each file
  */
-function checkForFocusedCases(elements: brs.types.BrsType[]) {
-    // iterate through elements and see if there are focused cases
-    for (let output of elements) {
-        if (output instanceof brs.types.RoAssociativeArray) {
-            let maybeFiles = output.getValue().get("fileswithfocusedcases");
-            let focusedFiles =
-                maybeFiles instanceof brs.types.RoArray
-                    ? maybeFiles.getElements()
-                    : [];
-            if (focusedFiles.length > 0) {
-                let formattedList = focusedFiles
-                    .map((brsStringPath) => `\t${brsStringPath.toString()}`)
-                    .join("\n");
-                console.error(
-                    c.red(
-                        `Error: used command line arg ${c.cyan(
-                            "--forbid-focused"
-                        )} but found focused tests in these files:\n${formattedList}`
-                    )
-                );
-
-                process.exitCode = 1;
-                return;
+async function getTestFiles(execute: ExecuteWithScope) {
+    let testsPattern = path.join(
+        process.cwd(),
+        `{test,tests,source,components}`,
+        "**",
+        "*.test.brs"
+    );
```

Comment:
Watch out!  Glob [only supports forward-slashes](https://github.com/isaacs/node-glob/tree/f5a57d3d6e19b324522a3fa5bdd5075fd1aa79d1#windows), so `path.join` will create `\` characters that `glob` thinks are escape characters!  Consider using either a regular template literal or `path.unix.join()` if the `join` model is your preference.